File: Hackerrank/sparse-arrays.py
N = int(raw_input())
strings = []
while len(strings) < N:
    p = raw_input()
    strings.append(p)
Q = int(raw_input())
queries = []
while len(queries)<Q:
    q = raw_input()
    queries.append(q)
count = 0
for t in queries:
    print(strings.count(t))

File: Hackerrank/sparse-arrays.cpp
#https://www.hackerrank.com/challenges/sparse-arrays/problem

vector<int> matchingStrings(vector<string> strings, vector<string> queries) {
    map<string,int> m;
    for(auto const& str:strings){
        m[str]++;
    }
    vector<int> v;
    for(auto const& x:queries){
        v.push_back(m[x]);
    }
    return v;
}